.github/workflows/lint.yml

```diff
@@ -0,0 +1,20 @@
+name: Rust Format Check
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  fmt:
+    name: fmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - name: Run fmt
+        run: cargo fmt --all --check
+        
```

Comment:
This is actually not an empty new line! I'd suggest making your editor trim trailing whitespace.

Reply:
yeah sorry I don't enable prettier for yml file, i've fixed